BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/RichPayloadEventSource.cs

```diff
@@ -259,7 +259,7 @@ public void Process(ITelemetry item)
                 CopyGlobalPropertiesIfRequired(item, telemetryItem.Properties);
                 item.Sanitize();
                 this.WriteEvent(
-                    AvailabilityTelemetry.TelemetryName,
+                    telemetryItem.EnvelopeName,
                     telemetryItem.Context.InstrumentationKey,
                     telemetryItem.Context.SanitizedTags,
                     telemetryItem.Data,
```

Comment:
would be great if this giant if/else could be consolidated into a single interface. :)
The only thing unique that i see is the Keyword, which could be put on the TelemetryItem as an internal field.